### DISCUSSION TITLE
Support custom request methods

### DISCUSSION THREAD

**Discussion by netclectic** [canton7/RestEase#184] [Q&A] [answered]
I'm trying to support an api that has a mix of standard rest and webdav, so I need to be able to make requests with methods like MKCOL, COPY and MOVE.

I thought I might be able to just create some custom request attributes, but the constructor for `RequestAttributeBase` is marked as internal. 

Is there some other way I can achieve this?

**Comment by canton7** [comment, marked as answer]
I was wondering when someone was going to ask this....

You can use `[Request("MKCOL")]` etc, would that fill your need?

The reason you can't inherit from `RequestAttributeBase` is the source generator backend: it can't actually instantiate your custom attribute, so it doesn't have a way of figuring out what `HttpMethod` it passes to the `RequestAttributeBase` ctor. It might be possible for the SG to emit something that runs at runtime to instantiate the attribute, but I didn't bother trying to implement something if noone was asking for it.

I short, if `RequestAttribute` works for you, great. If you need to be able to create your own attribute types, I can probably do it, but it's not supported currently.

**Comment by netclectic** [comment]
`[Request("MKCOL")]` works for me.

Thanks.
